drivers/usb_c/tcpc/ucpd_numaker.c

```diff
@@ -0,0 +1,2163 @@
+	NUMAKER_UTCPD_REG_DUMP(dev, FUTCTL);
+	NUMAKER_UTCPD_REG_DUMP(dev, PWRCTL);
+	NUMAKER_UTCPD_REG_DUMP(dev, CCSTS);
+	NUMAKER_UTCPD_REG_DUMP(dev, PWRSTS);
+	NUMAKER_UTCPD_REG_DUMP(dev, FUTSTS);
+	NUMAKER_UTCPD_REG_DUMP(dev, DVCAP1);
+	NUMAKER_UTCPD_REG_DUMP(dev, DVCAP2);
+	NUMAKER_UTCPD_REG_DUMP(dev, MSHEAD);
+	NUMAKER_UTCPD_REG_DUMP(dev, DTRXEVNT);
+	NUMAKER_UTCPD_REG_DUMP(dev, VBVOL);
+	NUMAKER_UTCPD_REG_DUMP(dev, SKVBDCTH);
+	NUMAKER_UTCPD_REG_DUMP(dev, SPDGTH);
+	NUMAKER_UTCPD_REG_DUMP(dev, VBAMH);
+	NUMAKER_UTCPD_REG_DUMP(dev, VBAML);
+	NUMAKER_UTCPD_REG_DUMP(dev, VNDIS);
+	NUMAKER_UTCPD_REG_DUMP(dev, VNDIE);
+	NUMAKER_UTCPD_REG_DUMP(dev, MUXSEL);
+	NUMAKER_UTCPD_REG_DUMP(dev, VCDGCTL);
+	NUMAKER_UTCPD_REG_DUMP(dev, ADGTM);
+	NUMAKER_UTCPD_REG_DUMP(dev, VSAFE0V);
+	NUMAKER_UTCPD_REG_DUMP(dev, VSAFE5V);
+	NUMAKER_UTCPD_REG_DUMP(dev, VBOVTH);
+	NUMAKER_UTCPD_REG_DUMP(dev, VCPSVOL);
+	NUMAKER_UTCPD_REG_DUMP(dev, VCUV);
+	NUMAKER_UTCPD_REG_DUMP(dev, PHYCTL);
+	NUMAKER_UTCPD_REG_DUMP(dev, FRSRXCTL);
+	NUMAKER_UTCPD_REG_DUMP(dev, VCVOL);
+	NUMAKER_UTCPD_REG_DUMP(dev, CLKINFO);
+
+	return 0;
+}
+
+static int numaker_eadc_vref_init(const struct device *dev)
+{
+	const struct numaker_tcpc_config *const config = dev->config;
+	struct numaker_tcpc_data *data = dev->data;
+	int rc = 0;
+	const struct adc_dt_spec *spec;
+	enum adc_reference reference;
+
+	if (data->vref_mv) {
+		goto cleanup;
+	}
+
+	/* NOTE: Register protection lock will restore automatically. Unlock it again. */
+	SYS_UnlockReg();
+
+	/* Analog reference voltage
+	 *
+	 * NOTE: For Vref being internal, external Vref pin must be floating,
+	 *       or it can disturb.
+	 */
+	spec = config->eadc.spec_vbus ? config->eadc.spec_vbus : config->eadc.spec_vconn;
+	if (spec) {
+		/* ADC device ready */
+		NUMAKER_RC_CHECK_WITH_MSG((adc_is_ready_dt(spec) ? 0 : -ENODEV),
+					  "ADC device for VBUS/VCONN not ready");
+		/* ADC channel configuration ready */
+		NUMAKER_RC_CHECK_WITH_MSG((spec->channel_cfg_dt_node_exists ? 0 : -ENODEV),
+					  "ADC channel configuration for VBUS/VCONN not specified");
+
+		reference = spec->channel_cfg.reference;
+
+		SYS->VREFCTL &= ~SYS_VREFCTL_VREFCTL_Msk;
+		if (reference == ADC_REF_EXTERNAL0 || reference == ADC_REF_EXTERNAL1) {
+			SYS->VREFCTL |= SYS_VREFCTL_VREF_PIN;
+		} else if (reference == ADC_REF_INTERNAL) {
+			switch (spec->vref_mv) {
+			case 1600:
+				SYS->VREFCTL |= SYS_VREFCTL_VREF_1_6V;
+				break;
+			case 2000:
+				SYS->VREFCTL |= SYS_VREFCTL_VREF_2_0V;
+				break;
+			case 2500:
+				SYS->VREFCTL |= SYS_VREFCTL_VREF_2_5V;
+				break;
+			case 3000:
+				SYS->VREFCTL |= SYS_VREFCTL_VREF_3_0V;
+				break;
+			default:
+				NUMAKER_RC_CHECK_WITH_MSG(-ENOTSUP, "Invalid Vref voltage");
+			}
+		} else {
+			NUMAKER_RC_CHECK_WITH_MSG(-ENOTSUP, "Invalid Vref source");
+		}
+
+		data->vref_mv = spec->vref_mv;
+	}
+
+cleanup:
+
+	return rc;
+}
+
+int numaker_utcpd_vbus_measure(const struct device *dev, uint32_t *mv)
```

Comment:
`static`, same for the ones below (unless I missed something)

Reply:
tcpc driver implements all the functions to be consistent with UTCPD H/W. These functions are to invoke by ppc and vbus drivers.